trace/kafka/kafka_test.go

```diff
@@ -160,3 +164,60 @@ func createKafkaBroker(t *testing.T, retError bool) *sarama.MockBroker {
 	seed.Returns(metadataResponse)
 	return seed
 }
+
+func TestSendWithCustomEncoder(t *testing.T) {
+	var u examples.User
+	firstname, lastname := "John", "Doe"
+	u.Firstname = &firstname
+	u.Lastname = &lastname
+	tests := []struct {
+		name        string
+		data        interface{}
+		key         string
+		enc         encoding.EncodeFunc
+		ct          string
+		wantMsgErr  bool
+		wantSendErr bool
+	}{
+		{name: "json success", data: "testdata1", key: "testkey1", enc: json.Encode, ct: json.Type, wantMsgErr: false, wantSendErr: false},
+		{name: "protobuf success", data: &u, key: "testkey2", enc: protobuf.Encode, ct: protobuf.Type, wantMsgErr: false, wantSendErr: false},
+		{name: "failure due to invalid data", data: make(chan bool), key: "testkey3", wantMsgErr: false, wantSendErr: true},
+		{name: "nil message data", data: nil, key: "testkey4", wantMsgErr: false, wantSendErr: false},
+		{name: "nil encoder", data: "somedata", key: "testkey5", ct: json.Type, wantMsgErr: false, wantSendErr: false},
+		{name: "json success", data: "", key: "testkey6", enc: json.Encode, ct: json.Type, wantMsgErr: false, wantSendErr: false},
+		{name: "json success", data: "", key: "🚖", enc: json.Encode, ct: json.Type, wantMsgErr: false, wantSendErr: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg, err := NewMessageWithKey("TOPIC", tt.data, tt.key)
+			if tt.key != "" {
+				assert.Equal(t, tt.key, *msg.key)
+			}
+			if tt.wantMsgErr == false {
+				assert.NoError(t, err)
+			} else {
+				assert.Error(t, err)
+			}
+
```

Comment:
do we need all this logic in the test ?
The changes are only in newAsyncProducer and createMessage.
Maybe we only need to cover those 2. 
(Also, should we maybe also test the error conditions from the NewAsyncProducer call ?)

Reply:
I agree with the first point; we don't need to test message creation here as well.

In regards with the second point, the NewAsyncProducer already has a few unit tests above; a PR is coming soon which will implement an AsyncProducer builder, and more tests will be written there.